content/components/text-input.mdx

```diff
@@ -0,0 +1,161 @@
+
+### Best practices
+
+- Only use placeholder text to show an example of the expected input. Never use placeholder text for critical information.
+- Use the width of the text input to give users a hint about how long the value is expected to be
+- If your field accepts free-form text _or_ an option from a predetermined list, consider using an [autocomplete input](/components/autocomplete)
+
+## Anatomy
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224447079-f95c75d2-5a3a-4175-bf9f-8725ad127a08.png"
+/>
+
+**Label (required):** The input's title. It should be as concise as possible and convey the purpose of the input. The label may be visually hidden in rare cases, but a label must be defined for assistive technologies such as a screen reader.
+
+**Required indicator:** Indicates that a value is required. Must be shown for any required field, even if all fields in the form are required.
+
+**Input (required):** Accepts user free-form text input
+
+**Leading visual:** An icon or short piece of text to reinforce the purpose of the input. Appears before the input's value.
+
+**Trailing visual:** The same as a leading visual, but appears after the input's value.
+
+**Trailing action:** An action that affects the input. Most commonly used for a button that clears the input.
+
+**Caption:** Provides additional context about the field to help users fill in the correct data or explain how the data will be used. Caption text should be as short as possible. Caption text may be displayed at the same time as a validation message, or it may be hidden if it only provides redundant information.
+
+**Validation message:** A single validation message may be displayed to provide helpful information for a user to complete their task. It is most commonly used to explain why a value is invalid so they can correct it and submit the form.
+
+## Options
+
+### Sizes
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445197-4b71262c-e1cc-4495-88d5-e46c6474d3cc.png"
+/>
+
+A text input can be made smaller or larger depending on the context it's being used in.
+
+You could make an input smaller if it is being used in a tight space with other small elements. You could make an input larger if it's in an open space or it feels out of proportion with adjacent large elements.
+
+### Width
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445198-7217e528-ed15-4fe9-aba9-264c7bc62e1b.png"
+/>
+
+By default, a text input will be given a default width by the browser.
+
+However, you can render a text input to fill the width of it's parent element, or set the width to whatever makes sense for your use-case. When setting a custom width, consider the length of the expected value.
```

Comment:
```suggestion
However, you can render a text input to fill the width of its parent element, or set the width to whatever makes sense for your use-case. When setting a custom width, consider the length of the expected value.
```

Reply:
Oops, this PR was supposed to _only_ contain changes relevant to TextInputWithTokens. Once we merge https://github.com/primer/design/pull/428, there should be way less changes in `text-input.md`.